gen/definitions/discovery.yaml

```diff
@@ -0,0 +1,203 @@
+---
+name: Discovery
+rest_endpoint: /dna/intent/api/v1/discovery
+get_rest_endpoint: /dna/intent/api/v1/discovery/1/500
+res_description:
+  'After discovery resource has been created, the Catalyst Center would contain device entries (if discovered).
+  All the device entries can be subsequently obtained from data source `catalystcenter_network_devices`. Terraform
+  currently is not able to handle `for_each` from a data source that depends on any managed resource, therefore to work
+  around that limitation the `catalystcenter_discovery` can be placed in a different tfstate (root module) than
+  `catalystcenter_network_devices` when the latter is used as a source of `for_each`.
+  <p/>
+  The discovery resource does not support updates, it needs to be destroyed and re-created instead.'
+ds_description:
+  'This data source retrieves information about a discovery. It is however not intended to be used to retrieve
+  the devices discovered, so instead please use data source `catalystcenter_network_devices` that can read all existing
+  network devices, including the discovered ones.'
+skip_minimum_test: true
+no_update: true
+no_import: true
+get_from_all: true
+id_from_query_path: response
+max_async_wait_time: 600
+doc_category: Discovery
+attributes:
+  - model_name: cdpLevel
+    type: Int64
+    description: CDP level is the number of hops across neighbor devices.
+    exclude_test: true
+    exclude_example: true
+    requires_replace: true
+  - model_name: discoveryType
+    type: String
+    enum_values: [Single, Range, Multi Range, CDP, LLDP, CIDR]
+    mandatory: true
+    description: Type of Discovery.
+    example: Range
+    requires_replace: true
+  - model_name: enablePasswordList
+    type: StringList
+    # description: Enable Password of the devices to be discovered.
```

Comment:
In fact I wanted to comment out the entire attribute. I'm not sure about the semantics of these 3 lists. Should it be in fact a single list with nested attribs? Also I don't have a liberty to touch device credentials at all in my environment, so I cannot really play and see myself what is going on with these.

Suggestions?

Reply:
According to the API documentation it is a list of strings, so this would be the right type to use. I would suggest to leave it in and uncomment the descriptions. We will test this with an internal instance.

Reply:
done, I've added descriptions